src/components/contributions-table.jsx

```diff
@@ -114,7 +122,61 @@ class ContributionsTable extends React.Component {
       return 0;
     };
 
+    function parseElectionName(input) {
+      const [ename, ...rest] = input.split('-');
+
+      if (rest.length === 1) {
+        // If there is only one element after the split, assume it's the year
+        return { ename, month: 'november', year: rest[0] };
+      } else if (rest.length >= 2) {
+        // If there are two or more elements, assume the last is the year
+        // and the second-to-last is the month
+        return { ename, month: rest[rest.length - 2], year: rest[rest.length - 1] };
+      }
+      // Handle the case where there are no elements after the split
+      return { ename, month: 'november', year: undefined };
+    }
+
+    const electionCompare = (x, y) => {
+      const election1 = parseElectionName(x);
+      const election2 = parseElectionName(y);
+
+      if (election1.name > election2.name) {
+        return 1;
+      }
+      if (election1.name < election2.name) {
+        return -1;
+      }
+
+      if (election1.year > election2.year) {
```

Comment:
This is comparing strings. Not sure if there's any "gotchas" with > & < comparing strings in JS. Could be fine, I don't know. Just wanted to point it out in case you'd be concerned.

Reply:
Actually it occurs to me that we do not need to (or maybe should not) sort on name.  The names will be all the same.  I don't think that a IEC will report collections in two different jurisdictions, but if they did then we would want the order to be on the date, regardless of the jurisdiction.